util/llbutil/secretprovider/meta.go

```diff
@@ -0,0 +1,29 @@
+package secretprovider
+
+import (
+	"github.com/google/uuid"
+)
+
+type secretID struct {
+	name          string
+	localProject  string
+	remoteProject string
+}
+
+var secretIDs map[string]secretID
+
+// NewSecretID is used to store secretID metadata and returns a uuid
+// which can be used to lookup this metadata
+// This idirection is required due to earthly not being able to (easily) pass this data through buildkit
+func NewSecretID(name, localProject, remoteProject string) string {
```

Comment:
@vladaionescu any thoughts on this approach -- creating a UUID which is passed to buildkit, then using that UUID to get back to the secretID metadata which contains both the secret name, and the local or remote project path.

Reply:
Perhaps I wouldn't include the local project as part of the API here. I would only include the following:
* Detect the Earthly "project" - we don't have this notion today though so this can be left empty or not included for now
* Get the secret ID that the user provides (e.g. from `RUN --secret FOO=+secrets/<my-secret-id>`)

Using these two, we can create an ID that we use with Buildkit (this could be simply "<project>/<my-secret-id>", rather than a UUID). This constructed ID will end up being consistent across runs, in case caching is an issue. But even if it isn't, I think it's simpler rather than doing the UUID trick.

Reply:
> we don't have this notion today though so this can be left empty or not included for now

I don't see how this could work for a user who has two different projects such as:

```
~/project-a/Earthfile
RUN --secret FOO=+secrets/<my-secret-id>

~/project-a/.env
my-secret-id=password-a

~/project-b/Earthfile
RUN --secret FOO=+secrets/<my-secret-id>

~/project-b/.env
my-secret-id=password-b
```

Currently, for a user to work on project-a, one must: `cd ~/project-a && earthly +target` (or alternatively run `earthly --env-file=~/project-a/.env +target`).

Without having the `project` feature available, a user looses the ability to supply different passwords depending on the location of the Earthfile.

> This constructed ID will end up being consistent across runs, in case caching is an issue

Good catch on this point; unfortunately we're hitting this. Supplying a hash instead of UUID half-fixes the issue, but if a user moves the local path of a project, that will bust the cache :disappointed: 

Maybe we'll have to enforce global-level secrets via the custom secret provider until we complete the project name feature.

Reply:
UPDATE: we'll eventually provide a `$EARHTLY_PROJECT_NAME` as an environment variable to the user's `secret_provider` script.